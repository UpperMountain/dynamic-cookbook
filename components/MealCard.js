import React from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import { BlurView } from "expo";
var s = require("./Styles");

const ImageContainer = props => {
  let pad = props.width == 210 ? 4 : 0;
  return (
    <View
      style={{
        width: props.width,
        height: props.height,
        overflow: "hidden",
        backgroundColor: "white",
        borderRadius: 19
      }}
    >
      {props.children}
      <View
        style={{
          position: "absolute",
          zIndex: 1,
          top: props.height * 0.78 + pad,
          width: props.width,
          height: props.height * 0.22 - pad,
          backgroundColor: "rgba(255, 255, 255, .6)"
        }}
      >
        <BlurView
          tint={"light"}
          intensity={65}
          style={{
            width: props.width,
            height: props.height * 0.22 - pad,
            paddingLeft: 16,
            paddingTop: 4 + pad
          }}
        >
          <Text style={[s.body, { marginBottom: pad }]}>Pancakes</Text>
          <Text style={s.caption}>20min | serves 4</Text>
        </BlurView>
      </View>
    </View>
  );
};

export class MealCard extends React.Component {
  constructor(props) {
    super();
    this.state = {};
  }

  render() {
    return (
<<<<<<< HEAD
      <View style={style.shadow}>
=======
      <View style={s.shadow}>
>>>>>>> general progress
        <ImageContainer width={210} height={262}>
          <Image
            style={{ width: 210, height: 262 }}
            source={require("../assets/images/pancakes.jpg")}
          />
        </ImageContainer>
      </View>
    );
  }
}

export class SmallMealCard extends React.Component {
  constructor(props) {
    super();
    this.state = {};
  }

  render() {
    return (
<<<<<<< HEAD
      <View style={style.shadow}>
=======
      <View style={s.shadow}>
>>>>>>> general progress
        <ImageContainer width={142} height={191}>
          <Image
            style={{ width: 142, height: 191 }}
            source={require("../assets/images/pancakes.jpg")}
          />
        </ImageContainer>
      </View>
    );
  }
}
