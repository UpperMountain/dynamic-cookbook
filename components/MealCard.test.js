import React from "react";
import renderer from "react-test-renderer";
import { MealCard } from "./MealCard";

it("should render without exploding", () => {
  renderer.create(<MealCard />);
});

it("should pass snapshot tests", () => {
  const cmp = renderer.create(<MealCard />);
  expect(cmp.toJSON()).toMatchSnapshot();
});
