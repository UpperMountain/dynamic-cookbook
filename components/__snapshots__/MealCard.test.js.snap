// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass snapshot tests 1`] = `
<View
  style={
    Object {
      "elevation": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 19,
        "height": 262,
        "margin": 10,
        "overflow": "hidden",
        "width": 210,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 262,
          "width": 210,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, .6)",
          "height": 53.64,
          "position": "absolute",
          "top": 208.36,
          "width": 210,
          "zIndex": 1,
        }
      }
    >
      <ExponentBlurView
        intensity={65}
        style={
          Array [
            Object {
              "height": 53.64,
              "paddingLeft": 16,
              "paddingTop": 8,
              "width": 210,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
        tint="light"
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "raleway",
                "fontSize": 14,
              },
              Object {
                "marginBottom": 4,
              },
            ]
          }
        >
          Pancakes
        </Text>
        <Text
          style={
            Object {
              "color": "rgba(0, 0, 0, .65)",
              "fontFamily": "raleway",
              "fontSize": 12,
              "letterSpacing": 1,
            }
          }
        >
          20min | serves 4
        </Text>
      </ExponentBlurView>
    </View>
  </View>
</View>
`;
